lib/libsmb2.c

```diff
@@ -1213,8 +1213,13 @@ smb2_open_async_with_oplock_or_lease(struct smb2_context *smb2, const char *path
                         break;
         }
 
-        /* create options */
-        create_options |= SMB2_FILE_NON_DIRECTORY_FILE;
+        if (flags & O_DIRECTORY) {
```

Comment:
O_DIRECTORY is a linux only thing.
That is not a super big deal but we will have to add some

#ifndef O_DIRECTORY
#define O_DIRECTORY <some value that does not clash with any of the other flags on linux/osx/windows>
#endif 

to this file